pkg/token/issuertoken/issuertoken.go

```diff
@@ -29,12 +34,17 @@ func doRetries(call func() (*http.Response, error), retries int) (*http.Response
 			return nil, err
 		}
 
-		if resp.StatusCode != http.StatusInternalServerError || retries == 0 {
+		if !isStatusRetryable(resp.StatusCode) || retries == 0 {
 			break
 		}
 
+		if resp.StatusCode == http.StatusTooManyRequests {
```

Comment:
I think that we should sleep between retries for all allowable retries.

Reply:
Done